### PR TITLE
Promote kueue v0.3.0

### PR DESCRIPTION
**Pull request by alculquicondor**
sha obtained from:

```
docker inspect gcr.io/k8s-staging-kueue/kueue:v0.3.0|grep kueue
            "gcr.io/k8s-staging-kueue/kueue:v0.3.0"
            "gcr.io/k8s-staging-kueue/kueue@sha256:212ab77e32a7da99ba8a7670e37ab8b14a74f501226844d73f82b1dac3ada16d"
```

### CODE REVIEW

**Review by tenzen-y — COMMENTED** [review]
I verified: 

```shell
$ docker image inspect gcr.io/k8s-staging-kueue/kueue:v0.3.0 | grep sha256:212ab77e32a7da99ba8a7670e37ab8b14a74f501226844d73f82b1dac3ada16d
            "gcr.io/k8s-staging-kueue/kueue@sha256:212ab77e32a7da99ba8a7670e37ab8b14a74f501226844d73f82b1dac3ada16d"
```

/lgtm

**Comment by tenzen-y** [comment]
@alculquicondor We need to update the https://github.com/kubernetes/k8s.io/blob/main/registry.k8s.io/manifests/k8s-staging-kueue/promoter-manifest.yaml instead of `k8s.gcr.io/images/k8s-staging-kueue/images.yaml`.

> [k8s.gcr.io Image registry is frozen from April 2023. Please update the manifests in registry.k8s.io folder instead]

**Comment by alculquicondor** [comment]
oh, good catch. I will change the PR

**Comment by tenzen-y** [comment]
@alculquicondor 
Sorry for my bad.
Actually, we must change the https://github.com/kubernetes/k8s.io/blob/main/registry.k8s.io/images/k8s-staging-kueue/images.yaml.

**Comment by alculquicondor** [comment]
Updated and also left a comment in the old file

File: k8s.gcr.io/images/k8s-staging-kueue/images.yaml
Comment (tenzen-y) [inline k8s.gcr.io/images/k8s-staging-kueue/images.yaml:1]: @alculquicondor 
If we update this file, we can not merge this PR 😢

https://github.com/kubernetes/k8s.io/pull/5103#issuecomment-1499587551

Comment (alculquicondor) [reply k8s.gcr.io/images/k8s-staging-kueue/images.yaml:1]: ahhhh, missed that

**Pull request merged**